### PR TITLE
feat: Portkey integration with LangChain

### PR DESCRIPTION
**Pull request by noble-varghese**
- **Description:** 

This pull request introduces the Portkey integration to the Langchain repository, enhancing its production capabilities. Portkey offers a comprehensive set of tools and features to optimize the performance and reliability of AI applications.

#### Key Features of Portkey's Integration with Langchain:

1. **🚪 AI Gateway**:
    - **Automated Fallbacks & Retries**: Ensure your application remains functional even if a primary service fails.
    - **Load Balancing**: Efficiently distribute incoming requests among multiple models.
    - **Semantic Caching**: Reduce costs and latency by intelligently caching results.
2. **🔬 Observability**:
    - **Logging**: Keep track of all requests for monitoring and debugging.
    - **Requests Tracing**: Understand the journey of each request for optimization.
    - **Custom Tags**: Segment and categorize requests for better insights.
3. **📝 Continuous Improvement with User Feedback**:
    - **Feedback Collection**: Seamlessly gather feedback on any served request, be it on a generation or conversation level.
    - **Weighted Feedback**: Obtain nuanced information by attaching weights to user feedback values.
    - **Feedback Metadata**: Incorporate custom metadata with the feedback to provide context, allowing for richer insights and analyses.
4. **🔑 Secure Key Management**:
    - **Virtual Keys**: Portkey transforms original provider keys into virtual keys, ensuring your primary credentials remain untouched.
    - **Multiple Identifiers**: Ability to add multiple keys for the same provider or the same key under different names for easy identification without compromising security.

  - **Issue:** the issue # it fixes (if applicable),
  - **Dependencies:** 
  
This integration incorporates an external library named `portkey-ai`, which can be easily installed via pip. `portkey-ai` plays a crucial role in the seamless operation of Portkey Integration and is an essential component of its internal processes.

  - **Tag maintainer:**  @baskaryan,
  - **Twitter handle:** [PortkeyAI](https://x.com/PortkeyAI?s=20)

- **Additional Notes:**
We have conducted a cleanup process by removing obsolete notebooks and files that are no longer relevant following the integration changes.

### CODE REVIEW

**Comment by noble-varghese** [comment]
@baskaryan Can you help me with the workflows here. Please approve the workflows and I can ensure the all tests are passing.
Also your review would be greatly appreciated. Please take a look when you have a moment. Thank you! :)

**Comment by noble-varghese** [comment]
@hwchase17 @baskaryan The checks have passed except the vercel deployment. Can you review the PR at the earliest. 
Thank you :)

**Comment by noble-varghese** [comment]
Hi @eyurtsev, could you please approve the workflow runs to trigger the pipeline and review this PR

File: docs/extras/integrations/providers/portkey/index.md
Comment (baskaryan) [inline docs/extras/integrations/providers/portkey/index.md:1]: instead of deleting this should we just update it to refer to new integrations?

Comment (noble-varghese) [reply docs/extras/integrations/providers/portkey/index.md:1]: That sounds like a better approach. Thanks :)

**Comment by noble-varghese** [comment]
@baskaryan I have updated the branch. Can you take a look please ?

**Comment by noble-varghese** [comment]
@baskaryan Is it possible to update the PR if everything seems fine ?

**Comment by baskaryan** [comment]
Apologies for the slow review! Pr has some merge conflicts, happy to re-review if you'd like to resolve

**Pull request closed without merging**